examples/semver.peggy

```diff
@@ -0,0 +1,60 @@
+/**
+ * SemVer.org v2
+ * https://semver.org/spec/v2.0.0.html
+ * For unit tests see: https://github.com/dselman/peggy-semver
+ */
+semver
+  = versionCore:versionCore
+    pre:('-' @preRelease)?
+    build:('+' @build)?
+  {
+    return { versionCore, pre, build };
+  }
+
+versionCore
+  = major:$numericIdentifier '.' minor:$numericIdentifier '.' patch:$numericIdentifier
+  {
+    return {
+      major: parseInt(major, 10),
+      minor: parseInt(minor, 10),
+      patch: parseInt(patch, 10),
+    };
+  }
+
+preRelease
+  = head:$preReleaseIdentifier tail:('.' @$preReleaseIdentifier)*
+  {
+    return [head, ...tail];
+  }
+
+build
+  = head:$buildIdentifier tail:('.' @$buildIdentifier)*
+  {
+    return [head, ...tail];
+  }
+
+preReleaseIdentifier
+  = alphanumericIdentifier
+  / numericIdentifier
+
+buildIdentifier
+  = alphanumericIdentifier
+  / digit+
+
+alphanumericIdentifier
+  = digit* nonDigit identifierChar*
+
+numericIdentifier
+  = '0' / (positiveDigit digit*)
```

Comment:
Maybe this?  

```peg.js
numericIdentifier
  = num:$(positiveDigit digit*) { return parseInt(num) }
  / '0' ![0-9] { return 0 }
```

Reply:
TBH I don't understand how this works right now. `positiveDigit digit*` should return `['1', ['2', '3']]`, and `parseInt(['1', ['2', '3']], 10) = 1`.

Reply:
I'd suggest a bit different code, though:
```
numericIdentifier = num:$('0' / [1-9][0-9]*) { return parseInt(num); }
```

IMO you don't have to write a lot of code to make it more readable. "Positive digit" doesn't add any more information to `[1-9]`.

Reply:
Trying to keep it closer to https://semver.org/spec/v2.0.0.html I think.

Reply:
> TBH I don't understand how this works right now.

The magic in the `$` operator that returns the text from the input instead of results of subexpressions.

> "Positive digit" doesn't add any more information to [1-9].

Agreed, especially since this rule is used only once

Reply:
Oh, I missed `$` up above.

Reply:
Anyway, I didn't find a place in the spec that would limit numbers by any limit, so I suspect `parseInt` doesn't formally comply to the spec :/